// @tanstack/solid-table@8.7.4/build/lib/index.js
/**
 * solid-table
 *
 * Copyright (c) TanStack
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.md file in the root directory of this source tree.
 *
 * @license MIT
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var tableCore = require('@tanstack/table-core');
var solidJs = require('solid-js');
var store = require('solid-js/store');

// /** @jsxImportSource solid-js */
function flexRender(Comp, props) {
  if (!Comp) return null;
  if (typeof Comp === 'function') {
    return solidJs.createComponent(Comp, props);
  }
  return Comp;
}
function createSolidTable(options) {
  const resolvedOptions = solidJs.mergeProps({
    state: {},
    // Dummy state
    onStateChange: () => {},
    // noop
    renderFallbackValue: null,
    mergeOptions: (defaultOptions, options) => {
      return solidJs.mergeProps(defaultOptions, options);
    }
  }, options);
  const table = tableCore.createTable(resolvedOptions);
  const [state, setState] = store.createStore(table.initialState);
  solidJs.createComputed(() => {
    table.setOptions(prev => {
      return solidJs.mergeProps(prev, options, {
        state: solidJs.mergeProps(state, options.state || {}),
        // Similarly, we'll maintain both our internal state and any user-provided
        // state.
        onStateChange: updater => {
          // merging isn't required because stores shallow update
          setState(updater);
          options.onStateChange?.(updater);
        }
      });
    });
  });
  return table;
}

exports.createSolidTable = createSolidTable;
exports.flexRender = flexRender;
Object.keys(tableCore).forEach(function (k) {
  if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
    enumerable: true,
    get: function () { return tableCore[k]; }
  });
});
